src/vos/vos_obj.c

```diff
@@ -1866,23 +1887,33 @@ vos_obj_iter_aggregate(daos_handle_t ih, bool discard)
 }
 
 static int
-vos_obj_iter_delete(struct vos_iterator *iter, void *args)
+vos_obj_iter_process(struct vos_iterator *iter, vos_iter_proc_op op,
+		     void *args)
 {
 	struct vos_obj_iter *oiter = vos_iter2oiter(iter);
 
-	switch (iter->it_type) {
+	switch (op) {
+	case VOS_ITER_PROC_OP_DELETE:
+		switch (iter->it_type) {
+		default:
+			D_ASSERT(0);
+			return -DER_INVAL;
+		case VOS_ITER_DKEY:
+		case VOS_ITER_AKEY:
+		case VOS_ITER_SINGLE:
+			return obj_iter_delete(oiter, args);
```

Comment:
I think we should just call an API here that is specific to single value rather than hooking up the corrupt callbacks and such

Reply:
Suggestions? I don't see any single value APIs available at the vos object level ... unless we introduce a new api layer? Arrays have the api from evtree.h, but the btree apis are only generic (not knowing that it's a single value), right?